### PR TITLE
chore: Update dependencies

### PR DESCRIPTION
**Pull request by rhuss**
This PR sets the direct dependencies of kn to:

```
require (
	github.com/knative/serving v0.5.2
	github.com/knative/test-infra v0.0.0-20190423185229-691caf51f2bb
	github.com/mitchellh/go-homedir v1.1.0
	github.com/spf13/cobra v0.0.3
	github.com/spf13/pflag v1.0.3
	github.com/spf13/viper v1.3.1
	k8s.io/api kubernetes-1.12.6
	k8s.io/apimachinery kubernetes-1.12.6
	k8s.io/cli-runtime kubernetes-1.12.6
	k8s.io/client-go kubernetes-1.12.6
	sigs.k8s.io/yaml v1.1.0
)
````

I.e. an update to 0.5.2 of Knative serving and to the Kubernetes versions fitting to serving 0.5.2.

### CODE REVIEW

**Comment by rhuss** [comment]
The only actual change was only in `go.mod` and `pkg/serving/service.go` to change `Pinned` to `DeprecatedPinned`. The rest is `go mod	vendor`.

File: pkg/serving/service.go [hunk 1]
```diff
@@ -25,8 +25,8 @@ func GetConfiguration(service *servingv1alpha1.Service) (*servingv1alpha1.Config
 		return &service.Spec.RunLatest.Configuration, nil
 	} else if service.Spec.Release != nil {
 		return &service.Spec.Release.Configuration, nil
-	} else if service.Spec.Pinned != nil {
-		return &service.Spec.Pinned.Configuration, nil
+	} else if service.Spec.DeprecatedPinned != nil {
```
Comment (matzew) [inline pkg/serving/service.go]: :+1:

**Comment by rhuss** [comment]
@sixolet @cppforlife Sorry for bugging you, but can we decide what to do with this PR ? It's tedious to update it constantly (and its also lying here since 14 days).

'happy to close it, but how do you feel about updating to serving 0.5.2 (from 0.3.0 as it is now) as dependency ?

**Pull request merged**